tests/TestCase/Database/QueryTest.php

```diff
@@ -403,6 +405,37 @@ public function testSelectRightJoin()
         $result->closeCursor();
     }
 
+    public function testSelfRightJoin()
+    {
+        $this->skipIf(
+            $this->connection->driver() instanceof \Cake\Database\Driver\Sqlite,
+            'SQLite does not support RIGHT joins'
+        );
+        $this->loadFixtures('Comments');
+        $query = new Query($this->connection);
+        $comments = TableRegistry::get('Comments', [
+            'alias' => 'Comments',
+            'table' => 'comments',
+            'connection' => $this->connection
+        ]);
+
+        $latest = $query->select([
+                'u_id' => 'Comments.user_id',
+                'latest' => $query->func()->max('Comments.created')
+            ])->from(['Comments' => 'comments'])
+            ->group('Comments.user_id');
+
+        $actual = $comments->find()->rightJoin(
```

Comment:
Why is the right join relevant?

Reply:
In my original code I had a right join. I just kept it in the test.

Reply:
Ok. I've posted a comment on #11367 as I think you use case is not generating the correct SQL for the join conditions as right side values are treated as bound parameters.